lib/commands/cryptographic-identities.ts

```diff
@@ -602,11 +604,35 @@ export class ImportCryptographicIdentity implements ICommand {
 			}
 
 			if(!password && importType === CryptographicIdentityConstants.PKCS12CERTIFICATE) {
-				password = this.$prompter.getPassword("Certificate file password", { allowEmpty: true }).wait();
+				isPasswordRequired = true;
 			}
 
-			var targetFile = this.$fs.createReadStream(certificateFile);
-			var result = this.$server.identityStore.importIdentity(<any>importType, password, targetFile).wait();
+			var targetFile : any;
+			var result : Server.CryptographicIdentityData[];
+			var noErrorOccurred : boolean;
+
+			for (var i = 0; i < CryptographicIdentityConstants.MAX_ALLOWED_PASSWORD_ATTEMPTS; ++i) {
+				noErrorOccurred = true;
+				targetFile = this.$fs.createReadStream(certificateFile);
+				if (isPasswordRequired) {
+					password = this.$prompter.getPassword("Certificate file password", { allowEmpty: true }).wait();
+				}
+
+				try {
+					result = this.$server.identityStore.importIdentity(<any>importType, password, targetFile).wait();
+				} catch(err) {
+					noErrorOccurred = false;
+					this.$logger.error(err.message + " Verify that you are entering the correct file and password.");
+				}
+
+				if (noErrorOccurred) {
+					break;
+				}
+			}
+
+			if (!noErrorOccurred) {
+				this.$errors.failWithoutHelp("Exceeded maximuim number of import attempts.");
```

Comment:
Cannot import file. Verify that you have provided the correct file and password and try again.

You have reached the maximum number of authentication attempts for this operation.